scipy/interpolate/_ndbspline.py

```diff
@@ -0,0 +1,204 @@
+                                 f" {d} are inconsistent:"
+                                 f" got {self.c.shape[d]} coefficients for"
+                                 f" {len(td)} knots, need at least {n} for"
+                                 f" k={k}.")
+
+        dt = _get_dtype(self.c.dtype)
+        self.c = np.ascontiguousarray(self.c, dtype=dt)
+
+    def __call__(self, xi, nu=None, extrapolate=None):
+        """Evaluate the tensor product b-spline at `xi`.
+
+        Parameters
+        ----------
+        xi : array_like, shape(..., ndim)
+            The coordinates to evaluate the interpolator at.
+            This can be a list or tuple of ndim-dimensional points
+            or an array with the shape (num_points, ndim).
+        nu : array_like, optional, shape (ndim,)
+            Orders of derivatives to evaluate. Each must be non-negative.
+        extrapolate : bool, optional
+            Whether to exrapolate based on first and last intervals in each
+            dimension, or return `nan`. Default is to ``self.extrapolate``.
+
+        Returns
+        -------
+        values : ndarray, shape xi.shape[:-1] + self.c.shape[ndim:]
+            Interpolated values at xi
+        """
+        ndim = len(self.t)
+
+        if extrapolate is None:
+            extrapolate = self.extrapolate
+        extrapolate = bool(extrapolate)
+
+        if nu is None:
+            nu = np.zeros((ndim,), dtype=np.intc)
+        else:
+            nu = np.asarray(nu, dtype=np.intc)
+            if nu.ndim != 1 or nu.shape[0] != ndim:
+                raise ValueError("invalid number of derivative orders nu")
+
+        # prepare xi : shape (..., m1, ..., md) -> (1, m1, ..., md)
+        xi = np.asarray(xi, dtype=float)
+        xi_shape = xi.shape
+        xi = xi.reshape(-1, xi_shape[-1])
+        xi = np.ascontiguousarray(xi)
+
+        if xi_shape[-1] != ndim:
+            raise ValueError(f"Shapes: xi.shape={xi_shape} and ndim={ndim}")
+
+        # prepare k & t
+        _k = np.asarray(self.k)
+
+        # pack the knots into a single array
+        len_t = [len(ti) for ti in self.t]
+        _t = np.empty((ndim, max(len_t)), dtype=float)
+        _t.fill(np.nan)
+        for d in range(ndim):
+            _t[d, :len(self.t[d])] = self.t[d]
+
+        # tabulate the flat indices for iterating over the (k+1)**ndim subarray
+        shape = tuple(kd + 1 for kd in self.k)
+        indices = np.unravel_index(np.arange(prod(shape)), shape)
+        _indices_k1d = np.asarray(indices, dtype=np.intp).T
+
+        # prepare the coefficients: flatten the trailing dimensions
+        c1 = self.c.reshape(self.c.shape[:ndim] + (-1,))
+        c1r = c1.ravel()
+        # XXX: is c1r guaranteed to be C-contiguous? Cython code assumes it is.
+
+        # replacement for np.ravel_multi_index for indexing of `c1`:
+        _strides_c1 = np.asarray([_//c1.dtype.itemsize for _ in c1.strides], dtype=np.intp)
```

Comment:
This is a style issue, but it annoys me. `_` is usually used to denote a throwaway or unused variable. In this case, it's not. While it has no effect on how the code executes, I think of it's use as an antipattern.